ss/pebbledb/db.go

```diff
@@ -40,6 +42,9 @@ type Database struct {
 	storage *pebble.DB
 	// Earliest version for db after pruning
 	earliestVersion int64
+
+	// Map of stores that have been updated
+	storeKeyDirty sync.Map
```

Comment:
just a nit: a direct `map[]` usage with `sync.Mutex` is typically much faster than `sync.Map`.

Reply:
Will leave with `sync.Map` for now for simplicity. May update later